src/OpenTelemetry/Metrics/MetricPoint.cs

```diff
@@ -726,6 +808,41 @@ internal void TakeSnapshot(bool outputDelta)
                             sw.SpinOnce();
                         }
 
+                        break;
+                    }
+
+                case AggregationType.ExponentialHistogram:
+                case AggregationType.ExponentialHistogramWithMinMax:
+                    {
+                        var sw = default(SpinWait);
+                        while (true)
+                        {
+                            if (Interlocked.Exchange(ref this.exponentialBucketHistogram.IsCriticalSectionOccupied, 1) == 0)
+                            {
+                                // Lock acquired
+                                this.snapshotValue.AsLong = this.runningValue.AsLong;
+                                this.exponentialBucketHistogram.SnapshotSum = this.exponentialBucketHistogram.RunningSum;
+                                this.exponentialBucketHistogram.SnapshotMin = this.exponentialBucketHistogram.RunningMin;
```

Comment:
We could avoid updating `SnapshotMin`, `SnapShotMax`, `RunningMin`, `RunningMax` for `case AggregationType.ExponentialHistogram:`.